backend/backend/api/routes/room.py

```diff
@@ -0,0 +1,68 @@
+from functools import lru_cache
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, status
+
+from backend.common.room import Room
+from backend.common.member import RoomMember
+
+from backend.api.models.room import RoomId
+from backend.api.models.player import Player
+
+from backend.api.dependencies.room import rooms, get_room_by_id
+from backend.api.dependencies.player import get_player_by_id, get_other_by_id
+
+router = APIRouter(tags=["rooms"])
+
+
+@router.post("/rooms/{player_id}", status_code=status.HTTP_201_CREATED,
+             response_model=RoomId)
+async def create_room(player: Annotated[Player, Depends(get_player_by_id)],
+                      rooms: Annotated[dict[str, Room], Depends(rooms)]):
+    room = Room()
+
+    player = RoomMember(id=player.player_id,
+                        name=player.player_name)
+    player.leader = True
+    room.add_member(player)
+    rooms.update({room.room_id: room})
+    return RoomId(room_id=room.room_id)
+
+
+@router.post("/rooms/{room_id}/{player_id}",
+             status_code=status.HTTP_202_ACCEPTED)
+async def enter_room(room: Annotated[Room, Depends(get_room_by_id)],
+                     player: Annotated[Player, Depends(get_player_by_id)]):
+    player_in_room = room.members.get(player.player_id)
+    if player_in_room:
+        return {}
+
+    player = RoomMember(id=player.player_id,
+                        name=player.player_name)
+
+    room.add_member(player)
+
+    return RoomMember
+
+
+@router.post("/rooms/{room_id}/players",
+            status_code=status.HTTP_200_OK,
+            response_model=list[RoomMember])
+async def get_room_players(room: Annotated[Room, Depends(get_room_by_id)]):
+    return list(room.members.values())
+
+
+@router.put("/room_leader/{room_id}/leader",
+            status_code=status.HTTP_204_NO_CONTENT)
+async def update_room_leader(room: Annotated[Room, Depends(get_room_by_id)],
+                             player: Annotated[Player, Depends(get_player_by_id)],
+                             other: Annotated[Player, Depends(get_other_by_id)]):
```

Comment:
these would be query params and not body which is unusual for PUT request

Reply:
this will ruin all the point of the dependecies and make this function complacted again

Reply:
it Depends() (XD get it?) on whatever you want to achieve, IMO utilizing the `get_player` (that can be called upon) with verification whether the user is leader or not may look as neat as u expect it to be

Reply:
I am aware but the only choise I am left with here is doing a spefical dependency with a low of if's in it which you will not like or have if's inside of the request itself which again- you dont like 
with an addition of you wanting to move a entire verification to the requests we are left with a function with like five if staments- do we really want that?

Reply:
also I didn't find any source saying that using query parameters in put is a bad thing so I am very confused with you insisting on this

Reply:
After reading a little bit and thinking it through I think it should look differently then represented here, So I will appoint us to look upon another option that might be agreed upon us both
 
I'm not insisting, I'm reviewing and this is my *opinion* on the matter. you don't have to do anything I say, but when you won't at least present that change formally

Reply:
You can resolve